src/client/gui/lib/vm_details/terminal.dart

```diff
@@ -126,12 +212,11 @@ class VmTerminalState extends ConsumerState<VmTerminal> {
             const SizedBox(height: 12),
             OutlinedButton(
               style: buttonStyle,
-              onPressed: !vmRunning
-                  ? null
-                  : () => setState(() {
-                        terminal = Terminal(maxLines: 10000);
-                        initTerminal();
-                      }),
+              onPressed: vmRunning
```

Comment:
Well, that is the expected behavior, as far as I understood it. It should be like this:
- If the instance is not running, any newly opened tab will just show an unopened shell with that button. If you then start the instance, that button will become clickable, so you open a shell.
- If an instance is running and a new tab is opened, the shell will start automatically.
- If you close your last tab, a new one will be opened automatically.
I agree that this design can be a little confusing. Scott mentioned it as well. We should discuss this with the design team later.

Reply:
Yes, I think this needs some rethinking. The disabled "open shell" button is confusing. I understand it is greyed out, but that is not totally obvious in the colored background and it feels disappointing that you see the option but you can't choose it. Also, it is inconsistent with the CLI, where you can just shell a non-running instance and have it started. And then yes, I don't get that tabbing thing. In the future, I think we need to:

1) enable "open shell" for instances that we can start, show a hint on the rest (like deleted, unknown, and transitioning, i.e. starting, suspending, etc.)
2) new tabs open the shell automatically, but are only allowed if the previous tab is open (plus button grayed out otherwise)
3) quitting the shell process causes the tab to close (i.e. EOF = closing tab)
4) when the last tab is closed, it is replaced with the "open shell" tab
5) if the instance stops running, all tabs are closed (including suspension, as it would be difficult to guarantee that the SSH connection was valid alive when it came back up).

But yes, let's discuss this with design. I think this is one of the first things that we need to fix.